tests/records.test.ts

```diff
@@ -139,6 +271,89 @@ describe('Records', () => {
         );
       });
 
+      it('should not be able to buy a record with a reserved name specified', async () => {
```

Comment:
nit: break these up into different tests (e.g `should not be able to buy a record with a name shorter than 5 characters` and `should not be able to buy a record that is reserved for specific wallet`, e.g.`should be able to buy a reserved name that has expired`, etc. 

isolates these scenarios and helps identify specific issues